crates/sui-framework-build/src/compiled_package.rs

```diff
@@ -48,6 +48,12 @@ pub struct BuildConfig {
 }
 
 impl BuildConfig {
+    pub fn new_for_testing() -> Self {
+        let mut build_config: Self = Default::default();
+        build_config.config.install_dir = Some(tempfile::TempDir::new().unwrap().into_path());
```

Comment:
This leaks the temp directory, IIUC, right? (i.e. `into_path` takes the path out of the `TempDir` so that it won't get deleted when the guard is dropped).

Is that an OK thing to do at the scale/number of tests we're running?  If not, I'd recommend something like:


```
struct TestBuildConfig {
    install_dir: TempDir,
    build_config: BuildConfig,
}
```

Reply:
we leak temp dirs all the time in tests, has never caused a problem so far.